<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1105</width>
    <height>603</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="layoutDirection">
   <enum>Qt::LeftToRight</enum>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="MyOpenGLWidget" name="MyWidget" native="true">
    <property name="geometry">
     <rect>
      <x>9</x>
      <y>-4</y>
      <width>501</width>
      <height>551</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
      <horstretch>1</horstretch>
      <verstretch>1</verstretch>
     </sizepolicy>
    </property>
    <property name="layoutDirection">
     <enum>Qt::LeftToRight</enum>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="styleSheet">
     <string notr="true">MyOpenGLWidget
{
background-color:rgb(25, 25, 25);
}</string>
    </property>
   </widget>
   <widget class="QLabel" name="box_file_name">
    <property name="geometry">
     <rect>
      <x>655</x>
      <y>3</y>
      <width>441</width>
      <height>41</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QLabel {
background-color:rgb(25, 25, 25);
}</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="alignment">
     <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
    </property>
    <property name="wordWrap">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="download_file">
    <property name="geometry">
     <rect>
      <x>519</x>
      <y>6</y>
      <width>129</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Загрузить файл</string>
    </property>
   </widget>
   <widget class="QPushButton" name="But_color_back">
    <property name="geometry">
     <rect>
      <x>527</x>
      <y>52</y>
      <width>96</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Цвет фона</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_44">
    <property name="geometry">
     <rect>
      <x>676</x>
      <y>57</y>
      <width>40</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Линии</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="But_color_line">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>53</y>
      <width>111</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Цвет линий</string>
    </property>
   </widget>
   <widget class="QComboBox" name="line_attribute">
    <property name="geometry">
     <rect>
      <x>877</x>
      <y>53</y>
      <width>117</width>
      <height>32</height>
     </rect>
    </property>
    <property name="editable">
     <bool>false</bool>
    </property>
    <property name="currentText">
     <string>Сплошная</string>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <item>
     <property name="text">
      <string>Нет</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Сплошная</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Пунктир</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="label_46">
    <property name="geometry">
     <rect>
      <x>999</x>
      <y>57</y>
      <width>47</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Размер</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QSpinBox" name="size_line">
    <property name="geometry">
     <rect>
      <x>1059</x>
      <y>57</y>
      <width>41</width>
      <height>21</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_45">
    <property name="geometry">
     <rect>
      <x>659</x>
      <y>93</y>
      <width>59</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Вершины</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="But_color_vertex">
    <property name="geometry">
     <rect>
      <x>729</x>
      <y>89</y>
      <width>112</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Цвет вершин</string>
    </property>
   </widget>
   <widget class="QComboBox" name="vertex_attribute">
    <property name="geometry">
     <rect>
      <x>883</x>
      <y>89</y>
      <width>111</width>
      <height>32</height>
     </rect>
    </property>
    <property name="editable">
     <bool>false</bool>
    </property>
    <property name="currentText">
     <string>Круг</string>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <item>
     <property name="text">
      <string>Нет</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Круг</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Квадрат</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="label_47">
    <property name="geometry">
     <rect>
      <x>999</x>
      <y>93</y>
      <width>47</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Размер</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QSpinBox" name="size_vertex">
    <property name="geometry">
     <rect>
      <x>1059</x>
      <y>93</y>
      <width>41</width>
      <height>21</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_39">
    <property name="geometry">
     <rect>
      <x>604</x>
      <y>126</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Перемещение</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_36">
    <property name="geometry">
     <rect>
      <x>524</x>
      <y>156</y>
      <width>16</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>X</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="spin_mov_X">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>152</y>
      <width>61</width>
      <height>21</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-10000</number>
    </property>
    <property name="maximum">
     <number>10000</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_33">
    <property name="geometry">
     <rect>
      <x>612</x>
      <y>154</y>
      <width>16</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Y</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="spin_mov_Y">
    <property name="geometry">
     <rect>
      <x>625</x>
      <y>152</y>
      <width>61</width>
      <height>21</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-10000</number>
    </property>
    <property name="maximum">
     <number>10000</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_37">
    <property name="geometry">
     <rect>
      <x>700</x>
      <y>154</y>
      <width>16</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Z</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="spin_mov_Z">
    <property name="geometry">
     <rect>
      <x>710</x>
      <y>152</y>
      <width>61</width>
      <height>21</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-10000</number>
    </property>
    <property name="maximum">
     <number>10000</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_40">
    <property name="geometry">
     <rect>
      <x>896</x>
      <y>127</y>
      <width>65</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Вращение</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QDial" name="dial_rot_X">
    <property name="geometry">
     <rect>
      <x>792</x>
      <y>157</y>
      <width>80</width>
      <height>80</height>
     </rect>
    </property>
    <property name="maximum">
     <number>359</number>
    </property>
   </widget>
   <widget class="QDial" name="dial_rot_Y">
    <property name="geometry">
     <rect>
      <x>892</x>
      <y>158</y>
      <width>80</width>
      <height>80</height>
     </rect>
    </property>
    <property name="maximum">
     <number>359</number>
    </property>
   </widget>
   <widget class="QDial" name="dial_rot_Z">
    <property name="geometry">
     <rect>
      <x>995</x>
      <y>156</y>
      <width>80</width>
      <height>80</height>
     </rect>
    </property>
    <property name="maximum">
     <number>359</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_38">
    <property name="geometry">
     <rect>
      <x>806</x>
      <y>240</y>
      <width>16</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>X</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="spin_rot_X">
    <property name="geometry">
     <rect>
      <x>826</x>
      <y>238</y>
      <width>43</width>
      <height>21</height>
     </rect>
    </property>
    <property name="maximum">
     <number>359</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_41">
    <property name="geometry">
     <rect>
      <x>916</x>
      <y>242</y>
      <width>16</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Y</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="spin_rot_Y">
    <property name="geometry">
     <rect>
      <x>936</x>
      <y>240</y>
      <width>43</width>
      <height>21</height>
     </rect>
    </property>
    <property name="maximum">
     <number>359</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_42">
    <property name="geometry">
     <rect>
      <x>1016</x>
      <y>240</y>
      <width>16</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Z</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="spin_rot_Z">
    <property name="geometry">
     <rect>
      <x>1036</x>
      <y>238</y>
      <width>43</width>
      <height>21</height>
     </rect>
    </property>
    <property name="maximum">
     <number>359</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_43">
    <property name="geometry">
     <rect>
      <x>900</x>
      <y>276</y>
      <width>58</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Масштаб</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="spin_scale">
    <property name="geometry">
     <rect>
      <x>900</x>
      <y>300</y>
      <width>61</width>
      <height>21</height>
     </rect>
    </property>
    <property name="minimum">
     <double>0.100000000000000</double>
    </property>
    <property name="maximum">
     <double>100.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.500000000000000</double>
    </property>
    <property name="value">
     <double>1.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_48">
    <property name="geometry">
     <rect>
      <x>890</x>
      <y>340</y>
      <width>62</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Проекция</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QComboBox" name="projection_type">
    <property name="geometry">
     <rect>
      <x>860</x>
      <y>364</y>
      <width>142</width>
      <height>32</height>
     </rect>
    </property>
    <property name="editable">
     <bool>false</bool>
    </property>
    <property name="currentText">
     <string>Центральная</string>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <item>
     <property name="text">
      <string>Центральная</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Параллельная</string>
     </property>
    </item>
   </widget>
   <widget class="QPushButton" name="reset">
    <property name="geometry">
     <rect>
      <x>860</x>
      <y>440</y>
      <width>141</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Сбросить данные</string>
    </property>
   </widget>
   <widget class="QLabel" name="show_color_back">
    <property name="geometry">
     <rect>
      <x>629</x>
      <y>57</y>
      <width>21</width>
      <height>21</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="styleSheet">
     <string notr="true">QLabel {
background-color: black;
}</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="show_color_line">
    <property name="geometry">
     <rect>
      <x>846</x>
      <y>57</y>
      <width>21</width>
      <height>21</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QLabel {
background-color: black;
}</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="show_color_vertex">
    <property name="geometry">
     <rect>
      <x>849</x>
      <y>93</y>
      <width>21</width>
      <height>21</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QLabel {
background-color: black;
}</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QSlider" name="slider_mov_X">
    <property name="geometry">
     <rect>
      <x>554</x>
      <y>182</y>
      <width>22</width>
      <height>371</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-1000</number>
    </property>
    <property name="maximum">
     <number>1000</number>
    </property>
    <property name="pageStep">
     <number>1</number>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="QSlider" name="slider_mov_Y">
    <property name="geometry">
     <rect>
      <x>639</x>
      <y>182</y>
      <width>22</width>
      <height>371</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-1000</number>
    </property>
    <property name="maximum">
     <number>1000</number>
    </property>
    <property name="pageStep">
     <number>1</number>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="QSlider" name="slider_mov_Z">
    <property name="geometry">
     <rect>
      <x>723</x>
      <y>182</y>
      <width>22</width>
      <height>371</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-1000</number>
    </property>
    <property name="maximum">
     <number>1000</number>
    </property>
    <property name="pageStep">
     <number>1</number>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <zorder>label_44</zorder>
   <zorder>label_46</zorder>
   <zorder>size_line</zorder>
   <zorder>label_45</zorder>
   <zorder>vertex_attribute</zorder>
   <zorder>label_47</zorder>
   <zorder>size_vertex</zorder>
   <zorder>spin_rot_X</zorder>
   <zorder>spin_rot_Y</zorder>
   <zorder>spin_rot_Z</zorder>
   <zorder>label_48</zorder>
   <zorder>projection_type</zorder>
   <zorder>reset</zorder>
   <zorder>MyWidget</zorder>
   <zorder>download_file</zorder>
   <zorder>box_file_name</zorder>
   <zorder>label_33</zorder>
   <zorder>label_36</zorder>
   <zorder>label_37</zorder>
   <zorder>label_39</zorder>
   <zorder>dial_rot_X</zorder>
   <zorder>spin_mov_X</zorder>
   <zorder>spin_mov_Y</zorder>
   <zorder>spin_mov_Z</zorder>
   <zorder>label_40</zorder>
   <zorder>label_38</zorder>
   <zorder>dial_rot_Y</zorder>
   <zorder>label_41</zorder>
   <zorder>dial_rot_Z</zorder>
   <zorder>label_42</zorder>
   <zorder>label_43</zorder>
   <zorder>spin_scale</zorder>
   <zorder>But_color_vertex</zorder>
   <zorder>But_color_line</zorder>
   <zorder>But_color_back</zorder>
   <zorder>line_attribute</zorder>
   <zorder>show_color_back</zorder>
   <zorder>show_color_line</zorder>
   <zorder>show_color_vertex</zorder>
   <zorder>slider_mov_X</zorder>
   <zorder>slider_mov_Y</zorder>
   <zorder>slider_mov_Z</zorder>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1105</width>
     <height>24</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MyOpenGLWidget</class>
   <extends>QWidget</extends>
   <header location="global">myopenglwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
